src/dialogs/PanelPropertiesWindow.vala

```diff
@@ -23,6 +23,232 @@
+            // convert to readable size, decide on unit
+            string unit;
+            double size;
+            if (bytes >= 1000000000000) {
+                size = bytes/1000000000000.0;
+                unit = "TB";
+            }
+            else if (bytes >= 1000000000) {
+                size = bytes/1000000000.0;
+                unit = "GB";
+            }
+            else if (bytes >= 1000000) {
+                size = bytes/1000000.0;
+                unit = "MB";
+            }
+            else if (bytes >= 1000) {
+                size = bytes/1000.0;
+                unit = "kB";
+            }
+            else {
+                size = bytes;
+                unit = "byte";
+            }
+            // prevent silly .0 if first decimal is 0
+            return "%.1f".printf(size).replace(".0", "").replace(",0", "").concat(" ", unit);
+        }
+
+        private int64[] get_folderdata () {
+            // folder properties: count items
+            string cmd = "find \"" + this.filepath + "\"";
+            string output;
+            try {
+                GLib.Process.spawn_command_line_sync(cmd, out output);
+                string[] all_lines = output.split("\n");
+                string[] real_files = all_lines[1:all_lines.length -1];
+                int64 totalsize = 0;
+                foreach (string l in real_files) {
+                    File file = File.new_for_path(l);
+                    int64 size = file.query_info ("*", FileQueryInfoFlags.NONE).get_size();
+                    totalsize += size;
+                }
+                int lines = real_files.length;
+                return {lines, totalsize};
+            } 
+            catch (SpawnError e) {
+                return {-1, -1};
+            }
+        }
+
+        private string[] get_filetime () {
+            int64 last_edit = (int64) general_info.get_attribute_uint64(FileAttribute.TIME_MODIFIED);
+            int64 last_access = (int64) general_info.get_attribute_uint64(FileAttribute.TIME_ACCESS);
+            return {
+                new DateTime.from_unix_local(last_access).to_string(), 
+                new DateTime.from_unix_local(last_edit).to_string()
+            };
+        }
+
+        private void toggle_executable (Gtk.ToggleButton button) {
+            bool ex = button.get_active();
+            string cmd = "chmod +x \"" + this.filepath + "\"";
+            if (!ex) {cmd = "chmod -x \"" + this.filepath + "\"";}
+            try {
+                Process.spawn_command_line_async(cmd);
+            }
+            catch (SpawnError e) {
+                // nothing to be done
+            }
+        }
+
+        private Gtk.Image get_fileicon () {
+            // get file icon, fallback to icon name "unknown" on error (which exists!)
+            string icon_name = "unknown";
+            try {
+                icon_name = general_info.get_icon ().to_string().split(" ")[3];
```

Comment:
@Jacob-Vlijm, at least in elementary, when showing properties of a folder itemview, it crash. Reason: 
general_info.get_icon ().to_string() return just "folder". Could you take a look to solve it?

Reply:
that's odd, "folder" is correct, should should do the job. Will look tomorrow morning, ned to teach late.

Reply:
Oh, wait, I see, apparently, I was looking one step to far in your comment. Will include a check on the n- strings in the output.

Reply:
@spheras think I fixed it for elementary. Need to continue. I'll post analternative version, so you can test, after teaching :) late today or tomorrow.